channels/generic/websockets.py

```diff
@@ -243,33 +268,36 @@ class WebsocketMultiplexer(object):
     The opposite of the demultiplexer, to send a message though a multiplexed channel.
 
     The multiplexer object is passed as a kwargs to the consumer when the message is dispatched.
-    This pattern allows the consumer class to be independant of the stream name.
+    This pattern allows the consumer class to be independent of the stream name.
     """
 
     stream = None
     reply_channel = None
 
-    def __init__(self, stream, reply_channel):
+    def __init__(self, stream, reply_channel, json_encoder=None):
```

Comment:
Yeah, I see the problem here about propagating it; this is the one thing on my first scan through that I sort of didn't like but understood why it had to be like this. Arguably it's the fault of the existing code architecture for not being so great.

The problem is that this way does end up making `group_send` take the extra argument and you'd have to pass it directly through from the calling code, which is not ideal. I might prefer the whole "specifying the multiplexer subclass on the demultiplexer" thing in that case.